pallets/identity/src/benchmarking.rs

```diff
@@ -23,57 +23,144 @@ use polymesh_common_utilities::traits::identity::TargetIdAuthorization;
 use polymesh_primitives::{
     AuthorizationData, Claim, CountryCode, IdentityId, InvestorUid, Permissions, Scope, Signatory,
 };
+
 use schnorrkel::Signature;
 use sp_std::prelude::*;
 
+#[cfg(feature = "std")]
+use schnorrkel::Keypair;
+#[cfg(feature = "std")]
+use sp_core::{crypto::Pair as TPair, sr25519::Pair};
+#[cfg(feature = "std")]
+const SIGNING_CTX: &[u8] = b"substrate";
+
 const SEED: u32 = 0;
 pub fn uid_from_name_and_idx(name: &'static str, u: u32) -> InvestorUid {
     InvestorUid::from((name, u).encode().as_slice())
 }
 
-pub fn make_account_without_did<T: Trait>(
-    name: &'static str,
-    u: u32,
-) -> (T::AccountId, RawOrigin<T::AccountId>) {
-    let account: T::AccountId = account(name, u, SEED);
-    let origin = RawOrigin::Signed(account.clone());
-    let _ = balances::Module::<T>::make_free_balance_be(&account, 1_000_000.into());
-    (account, origin)
-}
+pub type SecretKey = [u8; 64]; // Only in sr25519
+pub type PublicKey = [u8; 32]; // Only in sr25519
 
-pub fn make_account<T: Trait>(
-    name: &'static str,
-    u: u32,
-) -> (T::AccountId, RawOrigin<T::AccountId>, IdentityId) {
-    let (account, origin) = make_account_without_did::<T>(name, u);
-    let uid = uid_from_name_and_idx(name, u);
-    let _ = Module::<T>::register_did(origin.clone().into(), uid, vec![]);
-    let did = Module::<T>::get_identity(&account).unwrap();
-    (account, origin, did)
+/// Helper class to create accounts and its DID to simplify benchmarks and UT.
+pub struct User<T: Trait> {
+    pub account: T::AccountId,
+    pub secret: SecretKey,
+    pub origin: RawOrigin<T::AccountId>,
+    uid: Option<InvestorUid>,
+    pub did: Option<IdentityId>,
 }
 
-pub fn make_cdd_account<T: Trait>(u: u32) -> (T::AccountId, RawOrigin<T::AccountId>, IdentityId) {
-    let (cdd_account, cdd_origin, cdd_did) = make_account::<T>("cdd", u);
-    T::CddServiceProviders::add_member(cdd_did).unwrap();
-    (cdd_account, cdd_origin, cdd_did)
+impl<T: Trait> User<T> {
+    /// Create an account based on `name` and `u` with 1_000_000 as free balance.
+    /// It also registers the DID for that account.
+    pub fn new(name: &'static str, u: u32) -> Self {
+        let mut user = Self::without_did(name, u);
```

Comment:
Not really important but you could apply the builder pattern to avoid `mut`. I personally think that would help in other benchmarks that construct users as well. Something like
```
let user: User<T> = UserBuilder::<T>::without_did(name, u)
    .with_uid(uid)
    .with_new_did()
    .build();
```
where `with_new_did` calls `register_did` and gets it with `get_identity`.

However this is just a cosmetic improvement. Feel free to ignore.

Reply:
@vkomenda I've added the `UserBuilder`.